pkg/table/tables/partition.go

```diff
@@ -1312,30 +1317,46 @@ func (t *partitionedTable) locatePartitionCommon(ctx expression.EvalContext, tp
 		} else {
 			idx, err = t.locateRangePartition(ctx, partitionExpr, r)
 		}
+		if err != nil {
+			return -1, err
+		}
+		pi := t.Meta().Partition
+		if pi.CanHaveOverlappingDroppingPartition() {
+			pid := pi.Definitions[idx].ID
+			for _, droppingDef := range pi.DroppingDefinitions {
+				if pid == droppingDef.ID {
+					// idx != -1, so the correct partition, but give an error, since it should
+					// not be written to! For read it can check the Overlapping partition and ignore the error.
+					// One should use the next non-dropping partition for range, or the default
+					// partition for list partitioned table with default partition, for read.
+					return idx, table.ErrNoPartitionForGivenValue.GenWithStackByArgs(fmt.Sprintf("matching a partition being dropped, '%s'", droppingDef.Name.String()))
```

Comment:
So if the query matches a dropping partition, it will return an error directly?

Reply:
Yes, so that writes to that partition (in that specific DDL State) will fail, i.e. being blocked.